commands.go

```diff
@@ -55,6 +56,29 @@ func (gdb *Gdb) Send(operation string, arguments ...string) (map[string]interfac
 	return result, nil
 }
 
+// SendCheckError works like Send, except that if the result returned by
+// gdb has class=error, SendCheckError returns a non-nil error value.
+func (gdb *Gdb) SendCheckError(operation string, arguments ...string) (map[string]interface{}, error) {
+	result, err := gdb.Send(operation, arguments...)
+
+	if err != nil {
+		return result, err
+	}
+
+	if result["class"] == "error" {
+		if payload, isMap := result["payload"].(map[string]interface{}); isMap {
```

Comment:
Maybe just stringify the payload here instead of the whole result when no message is present?

Reply:
I was going to object, but I just noticed that the 'payload' structure is introduced in go, so the payload entry contains all the data sent by gdb, if any. So indeed, no point in stringifying more than the payload. I will add a message, though, since otherwise there's a chance of a `"nil"` error message when no payload is present.